parquet/pqarrow/schema.go

```diff
@@ -514,8 +515,10 @@ func arrowFromFLBA(logical schema.LogicalType, length int) (arrow.DataType, erro
 	switch logtype := logical.(type) {
 	case schema.DecimalLogicalType:
 		return arrowDecimal(logtype), nil
-	case schema.NoLogicalType, schema.IntervalLogicalType, schema.UUIDLogicalType:
+	case schema.NoLogicalType, schema.IntervalLogicalType:
 		return &arrow.FixedSizeBinaryType{ByteWidth: int(length)}, nil
+	case schema.UUIDLogicalType:
+		return extensions.NewUUIDType(), nil
```

Comment:
This extension type is registered by default, but we currently have some tests failing when we try to explicitly unregister this type. If we want to respect the unregistration and fallback to FixedBinary then perhaps we could use `arrow.GetExtensionType()` to get the type instead of using the constructor directly.

Reply:
I Split this out to a separate PR as per @kou's suggestion, and I've made this change in it. It's a good point!